pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_grant_in_txn failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_grant_in_txn [failed](https://tanzanite.cluster.engflow.com/invocations/default/ea9d0068-3938-40fe-95c1-1b15bfa05892?testReportRun=1&testReportShard=23&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [285460a47358406e65df9ff3deb5a0764f58a2bf](https://github.com/cockroachdb/cockroach/commits/285460a47358406e65df9ff3deb5a0764f58a2bf):


```
=== RUN   TestLogic_grant_in_txn
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogic_grant_in_txn311010739
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[10:03:50] setting distsql_workmem='65693B';
[10:03:55] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 19 statements
[10:03:57] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 36 statements
[10:03:59] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 48 statements
[10:04:01] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 56 statements
[10:04:11] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn with config fakedist-disk: 56 tests, 0 failures
    logic.go:4202: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: error while processing
    logic.go:4202: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: 
        expected success, but found
        (XXA00) transaction committed but schema change aborted with error: (57014): job 1004895720878473217 could not be loaded: failed to get payload for job 1004895720878473217: job-info-get: failed to read query result: query execution canceled
        cancel_checker.go:80: in init()
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_grant_in_txn311010739
--- FAIL: TestLogic_grant_in_txn (35.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=23</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_grant_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42336

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_grant_in_txn [failed](https://tanzanite.cluster.engflow.com/invocations/default/f5768640-faf0-4136-aef7-c89829903ebe?testReportRun=1&testReportShard=23&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [e1b5321eb9ffa00c00f1b3e7ff724cf5cf41d90e](https://github.com/cockroachdb/cockroach/commits/e1b5321eb9ffa00c00f1b3e7ff724cf5cf41d90e):


```
=== RUN   TestLogic_grant_in_txn
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogic_grant_in_txn1844251829
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[10:49:21] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 31 statements
[10:49:23] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 47 statements
[10:49:25] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 54 statements
[10:49:35] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn with config fakedist-disk: 56 tests, 0 failures
    logic.go:4218: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: error while processing
    logic.go:4218: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: 
        expected success, but found
        (XXA00) transaction committed but schema change aborted with error: (57014): polling for queued jobs to complete: poll-show-jobs: failed to read query result: query execution canceled
        cancel_checker.go:80: in init()
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_grant_in_txn1844251829
--- FAIL: TestLogic_grant_in_txn (28.23s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=23</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_grant_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_grant_in_txn [failed](https://tanzanite.cluster.engflow.com/invocations/default/77045a52-7976-44f8-8db9-ce511c020ef9?testReportRun=1&testReportShard=23&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [6a7fff7bfd54ccd845a19ff8c1a7d0174559e4df](https://github.com/cockroachdb/cockroach/commits/6a7fff7bfd54ccd845a19ff8c1a7d0174559e4df):


```
=== RUN   TestLogic_grant_in_txn
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogic_grant_in_txn3963621363
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[10:59:38] setting distsql_workmem='89760B';
[10:59:41] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 16 statements
[10:59:43] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 30 statements
[10:59:45] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 46 statements
[10:59:57] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn with config fakedist-disk: 56 tests, 0 failures
    logic.go:4218: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: error while processing
    logic.go:4218: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:172: 
        expected success, but found
        (XXA00) transaction committed but schema change aborted with error: (57014): job 1008304063236112385 could not be loaded: load-job-query: failed to read query result: query execution canceled
        cancel_checker.go:80: in init()
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_grant_in_txn3963621363
--- FAIL: TestLogic_grant_in_txn (30.80s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=23</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_grant_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/grant_in_txn
+++ b/pkg/sql/logictest/testdata/logic_test/grant_in_txn
@@ -1,6 +1,10 @@
 # This tests ensures that transactions to perform grants on entities created
 # inside of a transaction do not get blocked and take a very long time.
 
+# We've seen this test fail under race due to a timeout, which is likely
+# to happen in slower builds since the transaction is a large operation.
+skip under race
+
 statement ok
 SET statement_timeout = '10s';
 

